crates/ide_db/src/path_transform.rs

```diff
@@ -145,10 +151,60 @@ impl<'a> Ctx<'a> {
         match resolution {
             hir::PathResolution::TypeParam(tp) => {
                 if let Some(subst) = self.substs.get(&tp) {
-                    ted::replace(path.syntax(), subst.clone_subtree().clone_for_update().syntax())
+                    let parent = path.syntax().parent()?;
+                    if let Some(parent) = ast::Path::cast(parent.clone()) {
```

Comment:
Is there a more reliable (or nicer) way how to check if the substituted parameter has an associated type?

Reply:
Don't think so, at least not with how things are currently structured here